File: Python/251-missingNumber.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        length = len(nums) + 1
        summation = (length - 1) * length / 2
        return summation - sum(nums)

if __name__ == '__main__':
   s = Solution()
   print s.missingNumber([0])
   print s.missingNumber([0, 1])
   print s.missingNumber([0, 2])

File: C++/251-missingNumber.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int length = nums.size() + 1;
        int sum = (length * (length - 1)) / 2, realSum = 0;
        for (int i = 0; i < nums.size(); i++) {
             realSum += nums[i];
        };
        return sum - realSum;
    }
};